src/peergos/shared/user/fs/FileTreeNode.java

```diff
@@ -676,8 +676,7 @@ private FilePointer writableFilePointer() {
             byte[] newMapKey = new byte[32];
             random.randombytes(newMapKey, 0, 32);
             SymmetricKey ourBaseKey = this.getKey();
-            // a file baseKey is the key for the chunk, which hasn't changed, so this must stay the same
-            SymmetricKey newBaseKey = this.isDirectory() ? SymmetricKey.random() : ourBaseKey;
+            SymmetricKey newBaseKey = SymmetricKey.random();
```

Comment:
Can you explain this change?

Reply:
The basekey for a file is no longer also the encryption key, (that is now the data key). This gives us to freedom to rekey without re-encrypting the data.